src/use_promise_guard/index.es

```diff
@@ -0,0 +1,19 @@
+import useMounted from '../use_mounted';
+
+const usePromiseGuard = (deps) => {
+  const isMounted = useMounted(deps);
+
+  const execute = (getPromise) => new Promise((resolve, reject) => {
+    getPromise(resolve, reject)
+      .then((result) => {
+        if (isMounted.current) resolve(result);
+      })
+      .catch((result) => {
+        if (isMounted.current) reject(result);
+      });
+  });
+
+  return execute;
```

Comment:
Not sure if it's a good idea to use a higher order function.

I suggest following call:
```js
const safeFetchCar = usePromiseGuard([id], () => fetchCar(id));

// ... inside some click handler
safeFetchCar.then(car => setCar(car));
```